src/backend/src/game/game.controller.ts

```diff
@@ -22,43 +25,112 @@ export class GameController {
     private playersRepository: Repository<Player>,
     @InjectRepository(Round)
     private roundsRepository: Repository<Round>,
+    private roundService: RoundService,
+    private gameSseService: GameSseService,
+    private gameService: GameService,
   ) {}
 
+  @Get('sse')
+  async subscribePlayer(
+    @Body('playerId') playerId: number,
+    @Body('gameId') gameId: number,
+    @Res() res: Response,
+  ): Promise<void> {
+    const player = await this.playersRepository.findOne(playerId, {
+      relations: ['grabs'],
+    });
+    if (player.game.id != gameId) {
+      throw new BadRequestException(
+        `Player ${playerId} does not belong to game ${gameId}`,
+      );
+    }
+
+    res.set({
+      'Content-Type': 'text/event-stream',
+      'Cache-Control': 'no-cache',
+      'X-Accel-Buffering': 'no',
+      Connection: 'keep-alive',
+    });
+
+    await this.gameSseService.subscribeClient(res, {
+      playerId,
+      gameId,
+    });
+  }
+
   @Post('take')
   async take(
     @Body('playerId') playerId: number,
     @Body('howMany') howMany: number,
     @Body('timeTaken') timeTaken: number,
     @Body('roundId') roundId: number,
   ): Promise<number> {
-    const player = await this.playersRepository.findOne(playerId);
-    if (!player) {
-      throw new BadRequestException('Player does not exist');
-    }
+    const player = await this.playersRepository.findOne(playerId, {
+      relations: ['grabs'],
+    });
 
-    const round = await this.roundsRepository.findOne(roundId);
-    if (!round) {
-      throw new BadRequestException('Round does not exist');
-    }
+    let round = await Round.findOne(roundId, {
+      relations: ['playerMoves', 'game'],
+    }); // Check for updates
+    this.validatePlayerAndRound(player, round);
+
+    const grab = await Grab.create({
+      round,
+      player,
+      howMany,
+      timeTaken,
+    }).save();
+
+    round = await this.roundsRepository.findOne(roundId, {
+      relations: ['playerMoves', 'game'],
+    }); // Check for updates
```

Comment:
Do we know if we can get rid of this? I think we might be able to use round.reload()

Reply:
I tried using round.reload() and it doesn't update correctly? Like I log after round.reload() and after the first grab occurs, playerMoves.length is 0. So after the fourth grab, there are only 3 moves.

Reply:
And just to be sure that isn't fixed by awaiting round.reload()

Reply:
Yep, already have await round.reload()

Reply:
funky but alright